src/DynamicData/Binding/BindAndSortOptions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+namespace DynamicData.Binding;
+
+/// <summary>
+/// Options for bind the bind and sort operators.
+/// </summary>
+/// <param name="ResetThreshold">The reset threshold ie the number of changes before a reset is fired.</param>
+/// <param name="UseReplaceForUpdates"> When possible, should replace be used instead of remove and add.</param>
+/// <param name="ResetOnFirstTimeLoad"> Should a reset be fired for a first time load.This option is due to historic reasons where a reset would be fired for the first time load regardless of the number of changes.</param>
+/// <param name="UseBinarySearch"> Use binary search when the result of the comparer is a pure function.</param>
+/// <param name="InitialCapacity"> Set the initial capacity of the readonly observable collection.</param>
+public record struct BindAndSortOptions(
+    int ResetThreshold,
+    bool UseReplaceForUpdates,
+    bool ResetOnFirstTimeLoad,
+    bool UseBinarySearch,
+    int InitialCapacity)
+{
+    /// <summary>
+    /// Default bind and sort options.
+    /// </summary>
+    public static readonly BindAndSortOptions Default = new
```

Comment:
I would very much like us to move away from using non-obvious "special values". E.G. from context in the operator, it looks like you're using `0` as the value for `ResetThreshold` to mean "don't ever convert updates to resets". I think it's way clearer to just use `null` for that scenario. Similarly, `null` for `InitialCapacity` instead of `-1`, when the intention is to not set an initial capacity.

With those changes, I would say that a `Default` field here becomes superfluous, as all of the default values for these options then become the default value for the type itself. Unless you want to make the `Default` field non-readonly, so consumers can set their own defaults, as they see fit.

Reply:
Removed Defaut